### PR TITLE
(feature) Integration with MultiCluster CRDs

### PR DESCRIPTION
**Pull request by erickhernandezascencio**
## Checklist
- [X] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

Includes: 
- MultiClusterMode config flag
- Conversion from MCI -> networking.k8s.io/v1/Ingress
- Conversion MCS -> v1/Service

### CODE REVIEW

File: helm/ingress-azure/values-template.yaml [hunk 1]
```diff
@@ -20,6 +20,9 @@ kubernetes:
   # Port for AGIC's HTTP API endpoint
   httpServicePort: 8123
 
+  #Enables monitoring of MultiClusterIngress instead of Ingress resources
```
Comment (akshaysngupta) [inline helm/ingress-azure/values-template.yaml:23]: put space after #

File: helm/ingress-azure/values.yaml [hunk 2]
```diff
@@ -24,6 +24,9 @@ kubernetes:
   # Port for AGIC's HTTP API endpoint
   httpServicePort: 8123
 
+  #Enables monitoring of MultiClusterIngress instead of Ingress resources
```
Comment (akshaysngupta) [inline helm/ingress-azure/values.yaml:27]: space after #

File: pkg/controllererrors/errorcodes.go [hunk 3]
```diff
@@ -30,7 +30,8 @@ const (
 
 	// k8sContext package
 	ErrorEnpdointsNotFound              ErrorCode = "ErrorEnpdointsNotFound"
-	ErrorFetchingEnpdoints              ErrorCode = "ErrorFetchingEnpdoints"
+	ErrorFetchingEnpdoints              ErrorCode = "ErrorFetchingEndpoints"
```
Comment (akshaysngupta) [inline pkg/controllererrors/errorcodes.go:33]: can you correct ErrorFetchingEnpdoints to ErrorFetchingEndpoints. Seems like a spelling mistake :/

File: pkg/k8scontext/context.go [hunk 4]
```diff
@@ -286,20 +307,39 @@ func (c *Context) GetProhibitedTarget(namespace string, targetName string) *proh
...
-	for _, serviceInterface := range c.Caches.Service.List() {
-		service := serviceInterface.(*v1.Service)
-		if _, exists := c.namespaces[service.Namespace]; len(c.namespaces) > 0 && !exists {
-			continue
+	if IsInMultiClusterMode {
+		for _, multiClusterServiceInterface := range c.Caches.MultiClusterService.List() {
+			multiClusterService := multiClusterServiceInterface.(*multiClusterService.MultiClusterService)
+			service, exists := convert.FromMultiClusterService(multiClusterService)
+			if !exists {
+				continue
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go:315]: probably log an error

File: pkg/k8scontext/context.go [hunk 5]
```diff
@@ -322,8 +362,50 @@ func (c *Context) GetEndpointsByService(serviceKey string) (*v1.Endpoints, error
...
+		klog.Error(e.Error())
+		c.MetricStore.IncErrorCount(e.Code)
+		return nil, e
+	}
+
+	multiClusterService := multiClusterServiceInterface.(*multiClusterService.MultiClusterService)
+	endpoints := &v1.Endpoints{}
+	subset := v1.EndpointSubset{}
+
+	for _, globalEndpoint := range multiClusterService.Status.Endpoints {
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go:397]: globalEndpoint -> multiClusterEndpoint

File: pkg/k8scontext/context.go [hunk 6]
```diff
@@ -374,12 +456,27 @@ func (c *Context) IsEndpointReferencedByAnyIngress(endpoints *v1.Endpoints) bool
...
-		ingress, _ := convert.ToIngressV1(ingressInterface)
-		if _, exists := c.namespaces[ingress.Namespace]; len(c.namespaces) > 0 && !exists {
-			continue
+	if IsInMultiClusterMode {
+		klog.V(9).Infof("Fetching MultiCluster Ingresses")
+		for _, mciInterface := range c.Caches.MultiClusterIngress.List() {
+			mci := mciInterface.(*multiClusterIngress.MultiClusterIngress)
+			ingress, exists := convert.FromMultiClusterIngress(mci)
+			if !exists {
+				continue
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go:465]: log error as unable to convert

File: pkg/k8scontext/context.go [hunk 7]
```diff
@@ -439,6 +558,7 @@ func (c *Context) GetService(serviceKey string) *v1.Service {
 
 	service := serviceInterface.(*v1.Service)
 	return service
+
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go:561]: remove space

**Review by akshaysngupta — APPROVED** [review]

**Comment by arodriguezdlc** [comment]
Hello, is there any documentation about this feature? Thanks

**Pull request merged**